netebpfext/net_ebpf_ext.c

```diff
@@ -638,27 +700,47 @@ net_ebpf_extension_initialize_wfp_components(_Inout_ void* device_object)
 
     NET_EBPF_EXT_LOG_ENTRY();
 
-    if (_fwp_engine_handle != NULL) {
+    if (_wfp_engine_handle != NULL) {
         // already registered
         goto Exit;
     }
 
-    session.flags = FWPM_SESSION_FLAG_DYNAMIC;
-
-    status = FwpmEngineOpen(NULL, RPC_C_AUTHN_WINNT, NULL, &session, &_fwp_engine_handle);
+    // It is critical that this specific session be a static session. We explicitly ensure this by making sure that the
+    // FWPM_PROVIDER::flags member is zero.
+    // This logic ensures that we do not break WFP's processing rules for sessions and transactions when we create
```

Comment:
Simply quote [WFP object association](https://learn.microsoft.com/en-us/windows/win32/fwp/object-management#object-associations) to justify why we are keeping these session as static.